File: Dsa-day133.java
class Solution {
    public long maxSubarraySum(int[] nums, int k) {
        int n = nums.length;
        long[] prefix = new long[n + 1];
        for (int i = 0; i < n; ++i) {
            prefix[i + 1] = prefix[i] + nums[i];
        }
        
        long ans = Long.MIN_VALUE / 2;
        long[] modMin = new long[k];
        Arrays.fill(modMin, Long.MAX_VALUE / 2);
        
        for (int i = 0; i <= n; ++i) {
            int r = i % k;
            if (modMin[r] != Long.MAX_VALUE / 2) {
                ans = Math.max(ans, prefix[i] - modMin[r]);
            }
            modMin[r] = Math.min(modMin[r], prefix[i]);
        }
        
        return ans;
    }
}

File: Dsa-day133.py
class Solution:
    def maxSubarraySum(self, nums: List[int], k: int) -> int:
        n = len(nums)
        prefix = [0] * (n + 1)
        for i in range(n):
            prefix[i + 1] = prefix[i] + nums[i]
        
        ans = -10**18
        mod_min = [10**18] * k
        
        for i in range(n + 1):
            r = i % k
            if mod_min[r] != 10**18:
                ans = max(ans, prefix[i] - mod_min[r])
            mod_min[r] = min(mod_min[r], prefix[i])
        
        return ans